dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1459,8 +1434,49 @@ class CompileTest {
         throw Exception('Unsupported option for fake devices');
       case DeviceOperatingSystem.fuchsia:
         throw Exception('Unsupported option for Fuchsia devices');
+      case DeviceOperatingSystem.ios:
       case DeviceOperatingSystem.macos:
-        throw Exception('Unsupported option for macOS devices');
+        unawaited(stderr.flush());
+        late final String deviceId;
+        if(deviceOperatingSystem == DeviceOperatingSystem.ios) {
+          deviceId = 'ios';
+        } else if (deviceOperatingSystem == DeviceOperatingSystem.macos) {
+          deviceId = 'macos';
+        } else {
+          throw Exception('Attempted to run darwin compile workflow with $deviceOperatingSystem');
+        }
+
+        options.insert(0, deviceId);
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final Directory buildDirectory = dir(path.join(
+          cwd,
+          'build',
+        ));
+        final String? appBundlePath =
+            _findDarwinAppInBuildDirectory(buildDirectory.path);
+        if (appBundlePath == null) {
+          throw 'Failed to find app bundle in ${buildDirectory.path}';
+        }
+        final Directory appBundle = Directory(appBundlePath);
+        releaseSizeInBytes = await appBundle
+            .list(recursive: true)
+            .where((FileSystemEntity e) => e.runtimeType == File)
+            .fold<int>(0, (int totalSize, FileSystemEntity e) {
+              assert(e.runtimeType == File);
+              return totalSize + e.statSync().size;
+            });
+        if (reportPackageContentSizes) {
+          final Map<String, dynamic> sizeMetrics = await getSizesFromDarwinApp(
```

Comment:
I know the rest of this file uses `dynamic`, but you should avoid `dynamic` in null safe code.  Use `Object` or `Object`.